File: src/main/java/tk/avabin/tdg/beans/dtos/CharacterAttackDto.java
package tk.avabin.tdg.beans.dtos;

import lombok.Data;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Component;

import java.io.Serializable;

/**
 * Created by Avabin on 18.05.2017.
 */
@Component
@Data
@Scope("prototype")
public class CharacterAttackDto implements Serializable {
    private Integer id;
    private int ownerId;
    private int attackItemId;
    private Integer attackBonus;
    private String damage;
    private String crit;
    private Integer range;
    private String type;
    private String notes;
    private String ammunition;
    private Integer ammunitionCount;
}

File: src/main/kotlin/tk/avabin/tdg/beans/dtos/CharacterAttackDto.kt
package tk.avabin.tdg.beans.dtos

import org.springframework.context.annotation.Scope
import org.springframework.stereotype.Component
import java.io.Serializable

/**
 * Created by Avabin on 18.05.2017.
 */
@Component
@Scope("prototype")
data class CharacterAttackDto(
        var id: Int = 0,
        var name: String = "",
        var attackItem: ItemDto? = null,
        var attackBonus: Int = 0,
        var damage: String = "",
        var crit: String = "",
        var range: Int = 0,
        var type: String = "",
        var notes: String = "",
        var ammunition: String = "",
        var ammunitionCount: Int = 0
) : Serializable